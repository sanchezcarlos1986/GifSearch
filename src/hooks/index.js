export {default as useFetchGifs} from './useFetchGifs';
export {default as useForm} from './useForm';
