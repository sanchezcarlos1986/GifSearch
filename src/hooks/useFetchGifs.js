import {useState, useEffect} from 'react';
import {getGifs} from '~helpers';

/**
 * Represents a useEffect statement
 * @constructor
 * @param {string} category - Category to search and set data
 */
const useFetchGifs = category => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGifs(category).then(imgs =>
      setState({
        data: imgs,
        loading: false,
      }),
    );
  }, [category]);

  return state; // {data: [], loading: true}
};

export default useFetchGifs;
