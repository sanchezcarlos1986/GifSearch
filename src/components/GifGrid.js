import React from 'react';
import PropTypes from 'prop-types';
import {useFetchGifs} from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';

export default function GifGrid({category}) {
  const {data: images, loading} = useFetchGifs(category);

  return (
    <div className="GifGrid">
      <h3>{category}</h3>
      {loading && <p>Cargando...</p>}
      {images.length > 0 && (
        <ul>
          {images.map((img) => (
            <GifGridItem key={img.id} {...img} />
          ))}
        </ul>
      )}
    </div>
  );
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
