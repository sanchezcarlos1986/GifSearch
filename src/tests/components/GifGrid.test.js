import React from 'react';
import '@testing-library/jest-dom';
import GifGrid from '../../components/GifGrid';
import {shallow} from 'enzyme';
import {useFetchGifs} from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs.js');

describe('GifGrid', () => {
  let props;
  let component;

  it('should renders ok', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    props = {
      category: 'Mega Man',
    };
    component = shallow(<GifGrid {...props} />);
    expect(component).not.toBeNull();
    expect(component.hasClass('GifGrid')).toBe(true);
    expect(component).toMatchSnapshot();
  });

  it('should renders items when images are loaded', () => {
    const gifs = [
      {
        id: 'abc',
        url: 'https://mygif.com',
        title: 'totoro',
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    props = {
      category: 'Mega Man',
    };
    component = shallow(<GifGrid {...props} />);

    expect(component).not.toBeNull();
    expect(component.find('ul').exists()).toBe(true);
    expect(component.find('GifGridItem').length).toBe(gifs.length);
    expect(component.find('GifGridItem').prop('id')).toBe(gifs[0].id);
    expect(component.find('GifGridItem').prop('url')).toBe(gifs[0].url);
    expect(component.find('GifGridItem').prop('title')).toBe(gifs[0].title);
    expect(component).toMatchSnapshot();
  });
});
