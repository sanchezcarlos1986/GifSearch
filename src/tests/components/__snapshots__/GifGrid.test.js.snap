// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid should renders items when images are loaded 1`] = `
<div
  className="GifGrid"
>
  <h3>
    Mega Man
  </h3>
  <ul>
    <GifGridItem
      id="abc"
      key="abc"
      title="totoro"
      url="https://mygif.com"
    />
  </ul>
</div>
`;

exports[`GifGrid should renders ok 1`] = `
<div
  className="GifGrid"
>
  <h3>
    Mega Man
  </h3>
  <p>
    Cargando...
  </p>
</div>
`;
